### PR TITLE
 Hostgroup synchronization #93

### PR DESCRIPTION
**Pull request by secult**

### CODE REVIEW

**Comment by secult** [comment]
This should go to the next next (major), not to the next release.

File: api/mon/backends/zabbix/base.py [hunk 1]
```diff
@@ -1372,10 +1426,21 @@ def from_mgmt_data(cls, zapi, name):
         return container
 
     @staticmethod
-    def hostgroup_name_factory(dc_name, node_name='', vm_uuid='', tag=''):
-        name = ':{}:{}:{}:{}:'.format(dc_name, node_name, vm_uuid, tag)
+    def hostgroup_name_factory(hostgroup_name, dc_name=''):
+        if dc_name:
+            name = ':{}:{}:'.format(dc_name, hostgroup_name)
+        else:
+            name = dc_name
```
Comment (dn0) [inline api/mon/backends/zabbix/base.py]: Did you mean `name = hostgroup_name`?

File: gui/vm/forms.py [hunk 2]
```diff
@@ -245,6 +247,8 @@ class AdminServerSettingsForm(ServerSettingsForm):
                                       widget=ArrayWidget(tags=True, escape_space=False,
                                                          attrs={'class': 'tags-select2 narrow'}))
     monitoring_hostgroups = ArrayField(label=_('Monitoring hostgroups'), required=False, tags=True,
+                                       validators=[
```
[the remark this answers is not in the document]
Comment (secult) [reply gui/vm/forms.py]: It depends on the ArrayField.

File: api/mon/backends/zabbix/base.py (hunk 1 above)
Comment (secult) [reply api/mon/backends/zabbix/base.py]: "you had one job..."

**Comment by secult** [comment]
I'll also change the behavior of the visibility of the non-dc prefixed hostgroups.

File: api/mon/base/tasks.py [hunk 3]
```diff
@@ -64,14 +65,18 @@ def mon_template_list(task_id, dc_id, **kwargs):
 # noinspection PyUnusedLocal
 @cq.task(name='api.mon.base.tasks.mon_hostgroup_list', base=MgmtTask)
 @mgmt_task()
-def mon_hostgroup_list(task_id, dc_id, **kwargs):
+def mon_hostgroup_list(task_id, dc_id, dc_bound, *args, **kwargs):
     """
     Return list of hostgroups available in Zabbix.
     """
     dc = Dc.objects.get_by_id(int(dc_id))
+    if dc_bound:
```
[the remark this answers is not in the document]
Comment (secult) [reply api/mon/base/tasks.py]: The information in the task_id is not explicit.

**Pull request merged**